### PR TITLE
upgraded app.js with functional components

### PR DESCRIPTION
**Pull request by nobel10122025**
i have tried to convert the app.js from class component to functional component with Hooks.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,56 +1,45 @@
...
+    const [source , setSource] = useState('all')
+    const [language, setLanguage] = useState('all')
+    const [sort , setSort] = useState('all')
+    const [order , setOrder] = useState('all')
+    const [go_search, setGoSearch] = useState(false)
+    const [collapseFilters , setCollapseFilters] = useState(false)
+    const refText = useRef()
 
-    this.setState({ search: event.target.value });
+  function handle_change(event) {
```
Comment (Sanix-Darker) [inline src/App.js:18]:
```suggestion
  const handle_change = (event) => {
```

File: src/App.js [hunk 2]
```diff
@@ -1,56 +1,45 @@
-import React, { Component } from "react";
+import React, { useState,useRef, useEffect } from "react";
```
Comment (Sanix-Darker) [inline src/App.js:1]:
```suggestion
import React, { useState, useRef, useEffect } from "react";
```

**Review by Sanix-Darker — APPROVED** [review]
Thanks for this Pull Request 🙏🏽 !
looks Good to me!
Just a few propositions... i rather prefer `const` arrow functions than `functions`... declarations !

**Comment by nobel10122025** [comment]
i have committed all the suggestion .and thank you for approving

**Pull request merged**